staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go

```diff
@@ -1367,28 +1368,39 @@ type Condition struct {
 	// Many .condition.type values are consistent across resources like Available, but because arbitrary conditions can be
 	// useful (see .node.status.conditions), the ability to deconflict is important.
 	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=Available;Progressing;Degraded
```

Comment:
this is a regex in the validation code.

Reply:
Yeah, I based this on the comment above the type. The validation code looks like it [uses different regexes conditionally](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L42) which I think is more complex than the regex marker can do alone

Reply:
using an enum here is not correct... far more types than Available, Progressing, Degraded are supported

The regex validation seems to be conditional based on whether the type is qualified or not. I think that can be expressed like this:
```
(DNS1123_PATTERN/)?(QUALIFIED_NAME_PATTERN)
```

Reply:
@liggitt Ah, I'm not great with complex regexes so I wasn't aware if it could handle conditionals

Since dns1123 pattern is `dns1123LabelFmt + "(\\." + dns1123LabelFmt + ")*"` and qualified name fmt is `"(" + qnameCharFmt + qnameExtCharFmt + "*)?" + qnameCharFmt`, I tried substituting these into `(DNS1123_PATTERN/)?(QUALIFIED_NAME_PATTERN)` (evaluated out). This ended up being pretty complex from what I can see, not sure if there's a good way to simplify it